src/translator/service.cpp

```diff
@@ -137,6 +211,43 @@ AsyncService::AsyncService(const AsyncService::Config &config)
       logger_(config.logger) {
   ABORT_IF(config_.numWorkers == 0, "Number of workers should be at least 1 in a threaded workflow");
   workers_.reserve(config_.numWorkers);
+  // Initiate terminology map if present
+  if (!config_.terminologyFile.empty()) {
+    // Create an input filestream
+    std::ifstream myFile(config_.terminologyFile);
+
+    // Make sure the file is open
+    if(!myFile.is_open()) throw std::runtime_error("Could not open file: " + config_.terminologyFile);
+    std::string line;
+    while(std::getline(myFile, line)) {
+        // Create a stringstream of the current line
+        std::stringstream ss(line);
+
+        std::string srcword;
+        std::string replacementword;
+        getline(ss, srcword, '\t');
+        getline(ss, replacementword, '\n');
+        // @TODO it seems like removing the tags forces the model to copy which is
+        // I guess just as good and more reliable. In that case we just don't tell the model
+        // what the original source is and it just has no choice BUT to generate the target.
```

Comment:
Edit: ah you say it here explicitly. Copy is a copy with the assumption it won't try to translate because it doesn't know the translation. For Chinese <-> English I can imagine this working, but no way that English <-> French would accept something like that… right?